jackson-modules/jackson-exceptions/src/test/java/com/baeldung/mappingexception/JacksonMappingExceptionUnitTest.java

```diff
@@ -48,4 +47,29 @@ public final void givenObjectHasNoAccessorsButHasVisibleFields_whenSerializing_t
         assertThat(dtoAsString, containsString("booleanValue"));
     }
 
+    @Test
+    public void givenJsonWithInvalidList_whenDeserializing_thenThrowException() throws JsonParseException, IOException {
+        final String json = "{\"name\":\"Netherlands\",\"cities\":{\"Amsterdam\", \"Tamassint\"}}";
+        final ObjectMapper mapper = new ObjectMapper();
+
+        Exception exception = assertThrows(JsonMappingException.class, () -> mapper.reader()
+            .forType(Country.class)
+            .readValue(json));
+
+        assertTrue(exception.getMessage()
+            .contains("Cannot deserialize value of type `java.util.ArrayList<java.lang.String>`"));
+    }
+
+    @Test
+    public void givenJsonWithvalidList_whenDeserializing_thenCorrect() throws JsonParseException, IOException {
```

Comment:
Also, the "v" of valid should be capitalized.

Reply:
Done